src/main/java/org/springframework/data/redis/core/convert/MappingRedisConverter.java

```diff
@@ -357,7 +378,118 @@ public void write(Object source, final RedisData sink) {
 		for (IndexedData indexedData : indexResolver.resolveIndexesFor(entity.getTypeInformation(), source)) {
 			sink.addIndexedData(indexedData);
 		}
+	}
+
+	protected void writePartialUpdate(PartialUpdate<?> update, RedisData sink) {
+
+		RedisPersistentEntity<?> entity = mappingContext.getPersistentEntity(update.getTarget());
+
+		write(update.getValue(), sink);
+		if (sink.getBucket().keySet().contains("_class")) {
+			sink.getBucket().put("_class", null); // overwrite stuff in here
+		}
+
+		if (update.isRefreshTtl() && !update.getPropertyUpdates().isEmpty()) {
+
+			Long ttl = entity.getTimeToLiveAccessor().getTimeToLive(update);
+			if (ttl != null && ttl > 0) {
+				sink.setTimeToLive(ttl);
+			}
+		}
+
+		for (PropertyUpdate pUpdate : update.getPropertyUpdates()) {
+
+			String path = pUpdate.getPropertyPath();
+
+			if (UpdateCommand.SET.equals(pUpdate.getCmd())) {
+
+				KeyValuePersistentProperty targetProperty = getTargetPropertyOrNullForPath(path, update.getTarget());
+
+				if (targetProperty == null) {
 
+					writeInternal(entity.getKeySpace(), pUpdate.getPropertyPath(), pUpdate.getValue(),
```

Comment:
add a type check and deny eg. setting `foo-bar` as `String` for a `Long` property.